lighthouse-core/gather/gather-runner.js

```diff
@@ -189,7 +190,29 @@ class GatherRunner {
     const driver = options.driver;
     const tracingData = {traces: {}};
 
-    passes = GatherRunner.instantiateGatherers(passes);
+    if (typeof options.url !== 'string' || options.url.length === 0) {
+      return Promise.reject(new Error('You must provide a url to the driver'));
+    }
+
+    if (typeof options.flags === 'undefined') {
+      options.flags = {};
+    }
+
+    if (typeof options.config === 'undefined') {
+      return Promise.reject(new Error('You must provide a config'));
+    }
+
+    // Default mobile emulation and page loading to true.
```

Comment:
I agree. This looks like an artifact of my refactoring perhaps, rather than something that _should_ be there.